recipes-app/iot2050-firmware-update/files/iot2050-firmware-update.tmpl

```diff
@@ -465,28 +472,31 @@ class ManagedFirmwareUpdate(FirmwareUpdate):
         # extract file path
         self.extract_path = "/tmp"
 
-    def update_firmware(self):
         # Get available firmwares by checking the board name and the os information
-        firmware_names = self._update_conf.firmware_filter(
+        self.firmware_names = self._update_conf.firmware_filter(
             self._board_info.board_name, self._board_info.os_info)
 
-        if len(firmware_names) <= 0:
-            raise UpgradeError("No firmware image could be updated on this board")
+    def check_firmware(self):
+        if len(self.firmware_names) <= 0:
+            raise UpgradeError("Board version must be newer than the minimal "
+                "version, no firmware image could be updated on this board!")
```

Comment:
What is the "minimal version"? Is that printed as well?

And what is actually meant by that in the context of firmware updates? Are we talking about PG1/PG2 here? Then it is likely not minimal but rather an exact match.

Reply:
Yes, it prints that too. Please check line 411.

Our current design is to specify a minimum version in the firmware package, and then compare the motherboard OS version to the minimum version to decide upgrading or not. Please check line 409.

Reply:
Ah, wording problem: We are not talking about the board version but the OS image version here - please correct that.

Reply:
Here is an example:
```
root@iot2050-debian:~# ./iot2050-firmware-update.tmpl IOT2050-FW-Update-PKG-V2.3_01.03.01.02.tar.xz
Current board: SIMATIC IOT2050 Basic PG2

Firmware requires a minimal version of V01.01.01, the current OS has I01.04.00.02-5-g4d02dd0-dirty.
Board version must be newer than the minimal version, no firmware image could be updated on this board!
```